core/src/syscall/host.rs

```diff
@@ -207,3 +207,37 @@ where
     None
   }
 }
+
+pub struct SyscallHostSpawn;
+
+impl SyscallHostSpawn {
+  pub const fn new() -> Self {
+    Self
+  }
+}
+
+impl<T> Syscall<T> for SyscallHostSpawn
+where
+  T: HostInterface,
+{
+  fn execute(&self, ctx: &mut SyscallContext<T>, arg1: u32, arg2: u32) -> Option<u32> {
+    // let athena_ctx = ctx
+    //   .rt
+    //   .context
+    //   .as_ref()
+    //   .expect("Missing Athena runtime context");
+
+    let len = arg2;
+    let blob: Vec<u8> = ctx
+      .slice(arg1, len as usize / 4)
+      .iter()
+      .flat_map(|&num| num.to_le_bytes().to_vec())
+      .collect();
+
+    // get value from host
+    let host = ctx.rt.host.as_mut().expect("Missing host interface");
+    host.borrow_mut().spawn(blob);
```

Comment:
What was the reason to use `Arc<RefCell<...>>`? 

`RefCell` bypasses Rust's borrow checker rules by doing borrow checks in runtime. It will panic if the value is already borrows.

`Arc` is meant to be used in multi-threaded code, but afaict it's only used on a single thread. Wouldn't `Rc` be better suited here?

Reply:
See #61. We inherited this from upstream. I _think_ the idea here is that, eventually, they'll support parallelization, and we might as well. We could replace this with `Rc` or just leave it for now.